packages/x-date-pickers/src/TimePicker/shared.tsx

```diff
@@ -109,7 +112,7 @@ export function useTimePickerDefaultizedProps<
     ...applyDefaultViewProps({
       views: themeProps.views,
       openTo: themeProps.openTo,
-      defaultViews: ['hours', 'minutes'] as TView[],
+      defaultViews: defaultViews as TView[],
```

Comment:
This would only fix cases, where default `views` values are used.
It would also not impact the `DateTimePicker` or any other place the `MultiSectionDigitalClock` is used. 🤔 
<img width="330" alt="Screenshot 2024-06-12 at 10 25 32" src="https://github.com/mui/mui-x/assets/4941090/915b3dd9-1555-4c8b-9471-161520df1ddc">